.template.Debian/docker-entrypoint.sh

```diff
@@ -0,0 +1,327 @@
+	fi
+
+	if [ -n "$MYSQL_USER" ] && [ -n "$MYSQL_PASSWORD" ]; then
+		mysql_note "Creating user ${MYSQL_USER}"
+		docker_process_sql --database=mysql <<<"CREATE USER '$MYSQL_USER'@'%' IDENTIFIED BY '$MYSQL_PASSWORD' ;"
+
+		if [ -n "$MYSQL_DATABASE" ]; then
+			mysql_note "Giving user ${MYSQL_USER} access to schema ${MYSQL_DATABASE}"
+			docker_process_sql --database=mysql <<<"GRANT ALL ON \`$MYSQL_DATABASE\`.* TO '$MYSQL_USER'@'%' ;"
+		fi
+
+		docker_process_sql --database=mysql <<<"FLUSH PRIVILEGES ;"
+	fi
+}
+
+_mysql_passfile() {
+	# echo the password to the "file" the client uses
+	# the client command will use process substitution to create a file on the fly
+	# ie: --defaults-file=<( _mysql_passfile )
+	if [ -n "$MYSQL_ROOT_PASSWORD" ]; then
+		cat <<-EOF
+			[client]
+			password="${MYSQL_ROOT_PASSWORD}"
+		EOF
+	fi
+}
+
+# Mark root user as expired so the password must be changed before anything
+# else can be done (only supported for 5.6+)
+mysql_expire_root_user() {
+	if [ -n "$MYSQL_ONETIME_PASSWORD" ]; then
+		docker_process_sql --database=mysql <<-EOSQL
+			ALTER USER 'root'@'%' PASSWORD EXPIRE;
+		EOSQL
+	fi
+}
+
+# check arguments for an option that would cause mysqld to stop
+# return true if there is one
+_mysql_want_help() {
+	local arg
+	for arg; do
+		case "$arg" in
+			-'?'|--help|--print-defaults|-V|--version)
+				return 0
+				;;
+		esac
+	done
+	return 1
+}
+
+_main() {
+	# if command starts with an option, prepend mysqld
+	if [ "${1:0:1}" = '-' ]; then
+		set -- mysqld "$@"
+	fi
+
+	# skip setup if they aren't running mysqld or want an option that stops mysqld
+	if [ "$1" = 'mysqld' ] && ! _mysql_want_help "$@"; then
+		mysql_note "Entrypoint script for MySQL Server ${MYSQL_VERSION} started."
+
+		# Load various environment variables
+		docker_setup_env "$@"
+		mysql_check_config "$@"
+		docker_create_db_directories
+
+		# If container is started as root user, restart as dedicated mysql user
+		if [ "$(id -u)" = "0" ]; then
+			mysql_note "Switching to dedicated user 'mysql'"
+			exec gosu mysql "$BASH_SOURCE" "$@"
+		fi
+
+		# If this is true then there's no database, and it needs to be initialized
+		if [ ! -d "$DATADIR/mysql" ]; then
+			docker_verify_minimum_env
+			docker_init_database_dir "$@"
+
+			mysql_note "Starting temporary server"
+			docker_temp_server_start "$@"
+			mysql_note "Temporary server started."
+
+			docker_setup_db
+			docker_process_init_files /docker-entrypoint-initdb.d/*
+
+			mysql_expire_root_user
+
+			mysql_note "Stopping temporary server"
+			docker_temp_server_stop
+			mysql_note "Temporary server stopped"
+
+			echo
+			mysql_note "MySQL init process done. Ready for start up."
+			echo
+		fi
+	fi
+	exec "$@"
+}
+
+# This checks if the script has been sourced from elsewhere.
+# If so we don't perform any further actions
+# https://unix.stackexchange.com/a/215279
+if [ "${FUNCNAME[${#FUNCNAME[@]} - 1]}" != 'source' ]; then
```

Comment:
doesn't work with centos bash
```
FUNCNAME: bad array subscript
```

Reply:
That's confusing. While we don't have to worry about centos here, is there a Percona Server image that isn't using `centos:7`. `FUNCNAME` as an array was added in [bash 3.0](https://git.savannah.gnu.org/cgit/bash.git/tree/CHANGES?id=3ba697465bc74fab513a26dea700cc82e9f4724e#n5577) and CentOS should be plenty new enough with `4.2.46`.  :confused:

After some digging I found that `FUNCNAME` works fine within a function (like I already have in the [postgres version of this PR](https://github.com/docker-library/postgres/pull/496/files#diff-60713fb19f6c50a2473835df63ff17f9R28-R31)), so we should just use that here too.

```console
$ docker run -it --rm centos:7 bash
[root@bcfd12483c5e /]# cat test.sh 
#!/bin/bash

_is_sourced() {
	echo 'inside: ['"${FUNCNAME[@]}"']'
	echo
	[ "${FUNCNAME[${#FUNCNAME[@]} - 1]}" == 'source' ]
}

echo 'outside: ['"${FUNCNAME[@]}"']'
echo
if _is_sourced; then
	echo 'it is sourced'
fi
[root@bcfd12483c5e /]# bash test.sh
outside: []

inside: [_is_sourced main]

[root@bcfd12483c5e /]# . test.sh
outside: []

inside: [_is_sourced source]

it is sourced
```